solvebio/resource/object.py

```diff
@@ -316,17 +418,41 @@ def objects(self, **params):
     def ls(self, **params):
         return self.objects(**params)
 
-    def query(self, query=None, **params):
-        """Shortcut to query the underlying Dataset object"""
-        from solvebio import Dataset
+    def __getattr__(self, name):
+        """Shortcut to access attributes of the underlying Dataset resource"""
+        from solvebio.resource import Dataset
+
+        # Valid override attributes that let Object act like a Dataset
+        valid_dataset_attrs = [
+            # query data
+            'query', 'lookup', 'beacon',
+            # transform data
+            'import_file', 'export', 'migrate',
+            # dataset meta
+            'fields', 'template', 'imports', 'commits',
+            # helpers
+            'activity', 'saved_queries'
+        ]
 
+        try:
+            return self[name]
+        except KeyError as err:
+            if name in valid_dataset_attrs and self.is_dataset:
+                return getattr(
+                    Dataset(self.dataset_id, client=self._client), name)
```

Comment:
why not getattr(self.dataset, name)

Reply:
we don't need to do the lookup to get the Dataset object in most cases. Wanted to avoid making the GET request if possible, which is why I used the instance.refresh() logic within those methods that needed them

Reply:
but then why not just *not* do the lookup in `def dataset`?